pydantic_ai_slim/pydantic_ai/ui/vercel_ai/_event_stream.py

```diff
@@ -71,12 +80,37 @@ def _json_dumps(obj: Any) -> str:
     return to_json(obj).decode('utf-8')
 
 
+def _extract_denied_tool_ids(messages: list[UIMessage]) -> set[str]:
```

Comment:
I think we can inline this into the method on the class, if we make it a `cached_property`